source/unix/openssl_platform_init.c

```diff
@@ -125,69 +55,29 @@ static int s_resolve_libcrypto_hmac(enum aws_libcrypto_version version, void *mo
     hmac_ctx_init_ex init_ex_fn = HMAC_Init_ex;
 
     /* were symbols bound by static linking? */
-    bool has_102_symbols = init_fn && clean_up_fn && update_fn && final_fn && init_ex_fn;
-    bool has_111_symbols = new_fn && free_fn && update_fn && final_fn && init_ex_fn && reset_fn;
-
-    if (version == AWS_LIBCRYPTO_NONE) {
-        if (has_102_symbols) {
-            version = AWS_LIBCRYPTO_1_0_2;
-            FLOGF("auto-resolving libcrypto 1.0.2");
-        } else if (has_111_symbols) {
-            version = AWS_LIBCRYPTO_1_1_1;
-            FLOGF("auto-resolving libcrypto 1.1.1");
-        } else {
-            /* not pre-linked, need to ask for a specific version */
-            return AWS_LIBCRYPTO_NONE;
-        }
-    }
-
-    if (has_102_symbols) {
-        FLOGF("found static libcrypto 1.0.2 HMAC");
-    }
-    if (has_111_symbols) {
-        FLOGF("found static libcrypto 1.1.1 HMAC");
-    }
+    bool has_awslc_symbols = new_fn && free_fn && update_fn && final_fn && init_fn && init_ex_fn && reset_fn;
 
     /* If symbols aren't already found, try to find the requested version */
-    /* when built as a shared lib, and multiple versions of openssl are possibly
-     * available (e.g. brazil), select 1.0.2 by default for consistency */
-    if (!has_102_symbols && version == AWS_LIBCRYPTO_1_0_2) {
-        *(void **)(&init_fn) = dlsym(module, "HMAC_CTX_init");
-        *(void **)(&clean_up_fn) = dlsym(module, "HMAC_CTX_cleanup");
-        *(void **)(&update_fn) = dlsym(module, "HMAC_Update");
-        *(void **)(&final_fn) = dlsym(module, "HMAC_Final");
-        *(void **)(&init_ex_fn) = dlsym(module, "HMAC_Init_ex");
-        if (init_fn) {
-            FLOGF("found dynamic libcrypto 1.0.2 HMAC symbols");
-        }
-    }
-
-    if (!has_111_symbols && version == AWS_LIBCRYPTO_1_1_1) {
+    /* when built as a shared lib, and multiple versions of libcrypto are possibly
+     * available (e.g. brazil), select AWS-LC by default for consistency */
+    if (!has_awslc_symbols && version == AWS_LIBCRYPTO_LC) {
         *(void **)(&new_fn) = dlsym(module, "HMAC_CTX_new");
         *(void **)(&reset_fn) = dlsym(module, "HMAC_CTX_reset");
         *(void **)(&free_fn) = dlsym(module, "HMAC_CTX_free");
         *(void **)(&update_fn) = dlsym(module, "HMAC_Update");
         *(void **)(&final_fn) = dlsym(module, "HMAC_Final");
         *(void **)(&init_ex_fn) = dlsym(module, "HMAC_Init_ex");
         if (new_fn) {
-            FLOGF("found dynamic libcrypto 1.1.1 HMAC symbols");
+            FLOGF("found dynamic libcrypto HMAC symbols");
```

Comment:
this should be "found dynamic aws-lc HMAC symbols" right?

Reply:
Yes, I will update that.